primitives/core/src/ecdsa.rs

```diff
@@ -546,25 +545,57 @@ impl Pair {
 
 	/// Sign a pre-hashed message
 	pub fn sign_prehashed(&self, message: &[u8; 32]) -> Signature {
-		let message = secp256k1::Message::parse(message);
-		secp256k1::sign(&message, &self.secret).into()
+		let message = libsecp256k1::Message::parse(message);
+		libsecp256k1::sign(&message, &self.secret).into()
 	}
 
 	/// Verify a signature on a pre-hashed message. Return `true` if the signature is valid
 	/// and thus matches the given `public` key.
 	pub fn verify_prehashed(sig: &Signature, message: &[u8; 32], public: &Public) -> bool {
-		let message = secp256k1::Message::parse(message);
-	
+		let message = libsecp256k1::Message::parse(message);
+
 		let sig: (_, _) = match sig.try_into() {
 			Ok(x) => x,
 			_ => return false,
 		};
-	
-		match secp256k1::recover(&message, &sig.0, &sig.1) {
+
+		match libsecp256k1::recover(&message, &sig.0, &sig.1) {
 			Ok(actual) => public.0[..] == actual.serialize_compressed()[..],
 			_ => false,
 		}
-	}	
+	}
+
+	/// Verify a signature on a message. Returns true if the signature is good.
+	/// Parses Signature using parse_overflowing_slice
+	pub fn verify_deprecated<M: AsRef<[u8]>>(sig: &Signature, message: M, pubkey: &Public) -> bool {
+		let message = libsecp256k1::Message::parse(&blake2_256(message.as_ref()));
+		let (sig, ri) = match parse_signature_deprecated(&sig.0) {
+			Ok(sigri) => sigri,
+			_ => return false
+		};
+		match libsecp256k1::recover(&message, &sig, &ri) {
+			Ok(actual) => pubkey.0[..] == actual.serialize_compressed()[..],
+			_ => false,
+		}
+	}
+}
+
+#[cfg(feature = "full_crypto")]
+fn parse_signature(
```

Comment:
Can we rename this to `parse_standard_signature` and `parse_overflowing_signature` and similarly for all the above `verify_` public functions? The priority here should be make sure that all library users get to deliberately choose whether they want standard or overflowing, so we shouldn't be using the same old name.

Reply:
Can't change the trait fn names, but I changed `verify_deprecated` to `verify_overflowing`

Reply:
That doesn't do much unfortunately. If you can't change the other name for `standard` then I suggest putting the `overflowing` one to deprecated.

I don't have huge problems keeping it as is, but we do need to be slightly careful here since misusing `overflowing` vs `standard` may be a source of trouble.

Alternatively, we can create a separate structure for the `OverflowingSignature` and rename the current one to `StandardSignature`. That would be the best for clarify, but just that it may be a lot of work -- so I guess it's optional.

Reply:
@sorpaas ok thanks. I will leave it as `verify_deprecated`, since it also coincides with the names of the other versioned host functions in `primitives/io/src/lib.rs`.